### DISCUSSION TITLE
Subsonic "Browse By Tags" leaks data

### DISCUSSION THREAD

**Discussion by troycarpenter** [ampache/ampache#4005] [Q&A]
I'm not sure if this is an actual bug yet or something in the way the system works.

For the Subsonic mobile clients I have tried, there is a setting for the server that says "Browse By Tags...Browse by tags instead of by folder structure."

With that setting "unchecked", everything seems to work as expected and the user only sees artists/albums/songs/etc as allowed by the configured catalog filter.

However, when this is enabled, the server will present to the Subsonic client pretty much ALL artists in the artist database when browsing the music library.  If the user selects an artist to which they do not have access, then the screen shows no data and the user does not have access to the content.

I would contend that the user should not have access to the full list of artists.  The list should be restricted as it is when this feature is not enabled.

My guess is that the request for content is different with that feature enabled than when it is not.  The request seems to be bypassing the catalog filter, at least to get an artist list.  Things like "recently added" appear to correctly be filtered and only show albums to which the user has access.

I have tried this with two Android clients:
DSub with the feature off: Works as expected and content is limited
DSub with the feature on: Exposes all available artists, but prevents access to disallowed content.
GoSONIC: This app only operates in tags mode, so it has the same behavior as DSub with the tags mode on.

(I discovered this anomaly when testing the GoSONIC app)

**Comment by troycarpenter** [comment]
Someone else mentioned Power Ampache 2, so I tried it.  It has a similar problem but not quite the same.  In the Library screen, I can see albums the current user is not supposed to see (including album covers).  There are no songs listed for the errant albums.  The playlist, songs, and artists tabs appear to be properly filtered.

I went back to the Ampache web interface as the same restricted  user and the errant albums do not appear in the browse albums list.

**Reply by lachlan-00** [reply]
i think power ampache 2 has a local database so it will cache them across users

**Comment by lachlan-00** [comment]
which calls are you getting with the unfiltered items? 

i think with tag browse it will be albumList2. getAblumList will be for file/folder browse. Most of the subsonic calls are not using a browse so they're probably not filtered correctly

**Reply by lachlan-00** [reply]
all these stats calls are probably the issue, some are filtered by catalogs but not the same ways as a browse.

![image](https://github.com/user-attachments/assets/cbec877a-86dd-43db-9437-1503fc032c39)

**Reply by troycarpenter** [reply]
I figure there's a difference on what's being queried with respect to "browse by tags".  I supposed that one or two queries were not getting passed to the filter before being presented to the user.

Is there something I can look at on my side to confirm anything for you?

**Reply by lachlan-00** [reply]
i've been able to generate similar states. i'll just need to do a bit more testing.

the way i find these out is a user with access to no catalogs should get nothing. makes it pretty easy but i'll also script something to test that with my build scripts.

I test the api and subsonic so i can look for errors like that and i've been meaning to get started on validating the correct data appears in the responses
